src/components/hoc/RestrictedRoute.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router';
+
+export default function RestrictedRoute({
+  isAllowed = false,
+  component: Component,
+  render: RenderedComponent,
+  redirectTo,
+  ...routeProps
+}) {
+  return (
+    <Route
+      {...routeProps}
+      render={(props) => {
```

Comment:
where are these props coming from? what are you actually hand down to the component?

Reply:
These are any props that come from above the Route. I will add some comments to make clearer. 👍